code/modules/ghostroles/spawner/base.dm

```diff
@@ -0,0 +1,132 @@
+/datum/ghostspawner
+	var/short_name = null
+	var/name = null
+	var/desc = null
+	var/welcome_message = null
+	var/list/tags = list() //Tags associated with that spawner
+
+	//Vars regarding the spawnpoints and conditions of the spawner
+	var/list/spawnpoints = null //List of the applicable spawnpoints (by name)
+	var/landmark_name = null //Alternatively you can specify a landmark name
+	var/max_count = 0 //How often can this spawner be used
+	var/count = 0 //How ofen has this spawner been used
+	var/req_perms = null //What permission flags are required to use this spawner
+	var/req_perms_edit = R_ADMIN
+	var/req_head_whitelist = FALSE //If a head of staff whitelist is required
+	var/req_species_whitelist = null //Name/Datum of the species whitelist that is required, or null
+	var/enabled = TRUE //If the spawnpoint is enabled
+	var/enable_dmessage = TRUE //The message to send to deadchat if the ghostspawner is enabled or TRUE for a default message
+	var/respawn_flag = null //Flag to check for when trying to spawn someone of that type (CREW, ANIMAL, MINISYNTH)
+	var/jobban_job = null //If this is set, then it will check if the user is jobbanned from a specific job. Otherwise it will check for the name of the spawner
+
+	//Vars regarding the mob to use
+	var/mob/spawn_mob = null //The mob that should be spawned
+	var/list/variables = list() //Variables of that mob
+	var/mob_name = FALSE //The name of that mob; If null prompts for it
+	var/mob_name_prefix = null //The prefix that should be applied to the mob (i.e. CCIAA, Tpr., Cmdr.)
+	var/mob_name_suffix = null //The suffix that should be applied to the mob name
+	
+
+//Return a error message if the user CANT see the ghost spawner. Otherwise FALSE
+/datum/ghostspawner/proc/cant_see(mob/user) //If the user can see the spawner in the menu
+	if(req_perms) //Only those with the correct flags can see restricted roles
+		if(check_rights(req_perms, show_msg=FALSE, user=user))
+			return FALSE //Return early and dont perform whitelist checks if staff flags are met
```

Comment:
```suggestion
			return null //Return early and dont perform whitelist checks if staff flags are met
```
I would return `null` over `FALSE` here. Semantics a bit, but it's more consistent: if you get non-null, it's an error object. null means no error object thus no error.

Reply:
Return False is intentional here, so I can work with 0 / text in the ui.